src/editor/object_settings.cpp

```diff
@@ -293,7 +296,35 @@ ObjectSettings::add_music(const std::string& text, std::string* value_ptr,
                           std::optional<std::string> default_value,
                           unsigned int flags)
 {
-  add_file(text, value_ptr, key, std::move(default_value), {".music"}, {"/music"}, false, flags);
+  auto generate_help_text_for_file = [](std::string path) -> std::string
+  {
+    std::string filename = FileSystem::basename(path);
+    std::unique_ptr<SoundFile> sound_file;
+    try {
+      sound_file = load_sound_file(path);
+    } catch (...) {
+      return "";
+    }
+
+    std::string& author = sound_file->m_author;
+    std::string& license = sound_file->m_license;
+    std::string& title = sound_file->m_title;
+
+    bool hasMusicMetadata = !title.empty() || !author.empty() || !license.empty();
+    if (!hasMusicMetadata) {
+      return "";
+    }
+
+    std::string title_or_filename = title.empty() ? filename : "\"" + title + "\""; // assumes path is just a filename
+
+    std::string help_text =
+        title_or_filename + (author.empty() ? "" : "\n by " + author)
+        + (license.empty() ? "" : "\nLicense: " + license);
```

Comment:
`fmt::format` should be used here to support translations.

``fmt::format(fmt::runtime(_("{} by {}")), title_or_filename, author)`` in one variable, if an author is provided. If not, only copy the title to this variable (by the way, if no info is provided, maybe nothing should be shown at all?).

``"\n" + fmt::format(fmt::runtime(_("License: {}")), license)`` in another variable, if license is provided. If not, keep empty.

Return the two variables concatenated.

Reply:
Hi @Vankata453, good catch: I updated the code to support translations. However, while I wrote the code, I discovered that there's an issue preventing "{}" substitutions from working in the current Git master version. I reported this issue in issue #2700.

I am currently working on code changes based on your recommendation. There are some differences I would lke to make to your suggestions. Notably, I would like to translate just the word License with `_("License") + ": {}"` to take advantage of an existing .po file msgid "License" translations. 

There may be an issue with _("{} by {}") msgid for some languages like Japanese, where it seems that the order of the first two arguments should be swapped, relative to the English message. For example, '"Piano Man" by Billy Joel' is Birī Joeru no `piano man' in Japanese (according to Google Translate). I'll lookup how to support argument swapping with tinygettext.

Currently, there is a newline separating the TITLE and the "by AUTHOR" parts. Do you know if _("{}\nby {}") works?

Reply:
> Notably, I would like to translate just the word License with `_("License") + ": {}"` to take advantage of an existing .po file msgid "License" translations.

Seems fine.

> Currently, there is a newline separating the TITLE and the "by AUTHOR" parts. Do you know if _("{}\nby {}") works?

I believe it should.